### ISSUE TITLE
A question in pretraining

### ISSUE THREAD

**Issue by Huangzhw0221** [sunlicai/MAE-DFER#12]
I met some problems during pretraining and finetuning.
1. I pretrained MAE-DFER on voxceleb2 with 2 × 2 × 10 local region size , and finetune on FERV39K , but only achieve 46.18 UAR and 35.12 WAR, I wonder if some hyperparameters are not set correctly. Can you give a pre-trained script, either 2 × 2 × 10 or 2 × 5 × 10， to help me reproduce the pretrain result?
2. I tried on raw Video MAE with some other attention mechanisms, and all these methods perform well during pretrain period (e.g. the loss only 0.32 when pretrain). However, the acc durining finetune could only reach 46.xx UAR on FERV39K  and 68.xx on DFEW. Have you met some problems like this, and is there any solution or hyperparameters may affect to accuracy?

**Comment by sunlicai** [comment]
Hi @Huangzhw0221,

1. I uploaded the pre-training script [here](https://github.com/sunlicai/MAE-DFER/tree/master/scripts/voxceleb2).
2. I did not meet your mentioned problem.  Which attention mechanism did you use?

**Comment by Huangzhw0221** [comment]
Thank you for your help! 
The attention mechanisms I used include original multi-head attention, multi-spatial attention, local-global attention (your work), and some modified attention. Most of them would reach a small loss after pretraining on Voxceleb for 100 epochs, and the image restoration effect is also quite good. However, if I finetune on downstream dataset with all these pretrained model, the acc growth is particularly slow, which is far inferior to your model.
For example, when using your pretrained model(checkpoint-49.pth given in this repo) and your script, the acc would reach 37.xx UAR in epoch 2, and reach 46.xx UAR in epoch 10, it's quite fast. But when I pretrain by myself, the model is the same, the acc could just reach 25.xx in epoch 2, and the performance is quite bad. So I asked if there is any solution or hyperparameter may affect to accuracy.
Finally, I'd like to try with your script and thanks for your help again!

**Comment by sunlicai** [comment]
Actually, I did not heavily tune the hyperparameters. One guess is that there might be bugs during your fine-tuning (e.g., mismatch between pre-training and fine-tuning). You'd better  have a double check. 

Besides, you can adapt the provided script to pre-train MAE-DFER in your setting and also fine-tune the pre-trained model to see where is wrong.

**Comment by Huangzhw0221** [comment]
Thanks! I saw "info_clean.csv" in your script, is there any other pretreatment on voxceleb?

**Comment by sunlicai** [comment]
No, I just removed several broken videos. You can see my preprocessing code for [VoxCeleb2](https://github.com/sunlicai/MAE-DFER/blob/master/preprocess/voxceleb2.py).
